--- pbsuite/jelly/Support.py
+++ pbsuite/jelly/Support.py
@@ -8,6 +8,7 @@
 from pbsuite.utils.FileHandlers import *
 
 from networkx import Graph
+from functools import cmp_to_key
 
 USAGE = """Support.py <alignmentFile> <gapInfo> <outFile>
 
@@ -421,7 +422,7 @@
         
         newReads = self.layout(anchor, list(reads))
         
-        newReads.sort(cmp=lambda x,y: x.qstart < y.qstart)
+        newReads.sort(key=cmp_to_key(lambda x,y: x.qstart < y.qstart))
         return newReads
     
     """
@@ -483,7 +484,7 @@
                 regionHolder.qend = newGuys[-1].qend
                 ret.extend(newGuys)
         
-        ret.sort(cmp=lambda x,y: x.qstart < y.qstart)
+        ret.sort(key=cmp_to_key(lambda x,y: x.qstart < y.qstart))
         return ret
 
     def isDiscordant(self, alignment, tailAllowed=150):
@@ -703,7 +704,7 @@
         readName = alignmentGroup[0].qname
         logging.debug("looking at: " + readName + " for scaffold extension/unification")
         
-        alignmentGroup.sort(cmp=lambda x,y: x.qstart - y.qstart)
+        alignmentGroup.sort(key=cmp_to_key(lambda x,y: x.qstart - y.qstart))
         anchor = self.alignCon.getBestScore(alignmentGroup)
         
         flags = []
