--- pbsuite/jelly/Assembly.py
+++ pbsuite/jelly/Assembly.py
@@ -33,7 +33,7 @@
     #input reads
     reads = FastqFile(query)
     #map to make the primary
-    primary= NamedTemporaryFile(prefix="primary_", suffix=".m4", delete=False, dir=basedir)
+    primary= NamedTemporaryFile(mode='w', prefix="primary_", suffix=".m4", delete=False, dir=basedir)
     primary = primary.name
     ALLTEMPFILES.append(primary)
     blasr(query, target, fmt="4", nproc=nproc, bestn=1, outname=primary)
@@ -49,7 +49,7 @@
     mapOut = open(outname, "w")
     
     #where I'm putting the tails
-    tfq = NamedTemporaryFile(prefix="tails_", suffix=".fastq", delete=False, dir=basedir)
+    tfq = NamedTemporaryFile(mode='w', prefix="tails_", suffix=".fastq", delete=False, dir=basedir)
     ALLTEMPFILES.append( tfq.name )
     whichEnd = defaultdict(list)
     #extract the tails
@@ -73,7 +73,7 @@
     tfq.close()
     logging.info("%d unmapped tails" % (ntails))
     #map tails
-    tailAlign = NamedTemporaryFile(prefix="tails_", suffix=".m5", delete=False, dir=basedir)
+    tailAlign = NamedTemporaryFile(mode='w', prefix="tails_", suffix=".m5", delete=False, dir=basedir)
     tailAlign = tailAlign.name
     ALLTEMPFILES.append(tailAlign)
     blasr(tfq.name, target, nproc=nproc, bestn=1, outname=tailAlign)
@@ -107,9 +107,9 @@
     returns queryFileName, targetFileName
     """
     global ALLTEMPFILES
-    query = NamedTemporaryFile(prefix="query_", suffix=".fastq", delete=False, dir=basedir)
+    query = NamedTemporaryFile(mode='w', prefix="query_", suffix=".fastq", delete=False, dir=basedir)
     ALLTEMPFILES.append(query.name)
-    target = NamedTemporaryFile(prefix="target_", suffix=".fasta", delete=False, dir=basedir)
+    target = NamedTemporaryFile(mode='w', prefix="target_", suffix=".fasta", delete=False, dir=basedir)
     ALLTEMPFILES.append(target.name)
     for read in reads:
         if read.startswith("ref"):
@@ -468,7 +468,7 @@
             logging.warning("fill sequence is small (%dbp) can't call consensus" % (len(bestSpan.seq)))
             #I don't know what to return here
             
-        ssfout = NamedTemporaryFile(prefix="span_", suffix=".fasta", delete=False, dir=basedir)
+        ssfout = NamedTemporaryFile(mode='w', prefix="span_", suffix=".fasta", delete=False, dir=basedir)
         ALLTEMPFILES.append(ssfout.name)
         logging.debug("spanning with %s" % (bestSpan.name))
         ssfout.write(">%s\n%s\n" % (bestSpan.name, bestSpan.seq))
@@ -483,7 +483,7 @@
         if len(bestF1E.seq) < 50:
             logging.warning("f1e sequence is small (%dbp) can't call consensus" % (len(bestF1E.seq)))
             #I don't know what to return here
-        f1sfout = NamedTemporaryFile(prefix="flank1_", suffix=".fasta", delete=False, dir=basedir)
+        f1sfout = NamedTemporaryFile(mode='w', prefix="flank1_", suffix=".fasta", delete=False, dir=basedir)
         ALLTEMPFILES.append(f1sfout.name)
         f1sfout.write(">%s\n%s\n" % (bestF1E.name, bestF1E.seq))
         f1sfout.close()
@@ -497,7 +497,7 @@
         if len(bestF2E.seq) < 50:
             logging.warning("f2e sequence is small (%dbp) can't call consensus" % (len(bestF2E.seq)))
             #I don't know what to return here
-        f2sfout = NamedTemporaryFile(prefix="flank2", suffix=".fasta", delete=False, dir=basedir)
+        f2sfout = NamedTemporaryFile(mode='w', prefix="flank2", suffix=".fasta", delete=False, dir=basedir)
         ALLTEMPFILES.append(f2sfout.name)
         f2sfout.write(">%s\n%s\n" % (bestF2E.name, bestF2E.seq))
         f2sfout.close()
@@ -519,7 +519,7 @@
     #try to build span
     if SUPPORTFLAGS.span in data.stats["support"][0]:
         logging.debug("build span")
-        alignFile = NamedTemporaryFile(prefix="scon_", suffix=".m5", delete=False, dir=args.tempDir)
+        alignFile = NamedTemporaryFile(mode='w', prefix="scon_", suffix=".m5", delete=False, dir=args.tempDir)
         alignFile.close(); alignFile = alignFile.name
         ALLTEMPFILES.append(alignFile)
         #blasr(data.spanReads, data.spanSeed, bestn = 1, nproc = args.nproc, outname=alignFile)
@@ -558,7 +558,7 @@
     logging.debug((fl1Flag, fl2Flag))
     if fl1Flag in data.stats["support"][1]:
         logging.debug("build flank1 %d" % fl1Flag)
-        alignFile = NamedTemporaryFile(prefix="f1con_", suffix=".m5", delete=False, dir=args.tempDir)
+        alignFile = NamedTemporaryFile(mode='w', prefix="f1con_", suffix=".m5", delete=False, dir=args.tempDir)
         alignFile.close(); alignFile = alignFile.name
         ALLTEMPFILES.append(alignFile)
         #blasr(data.flank1Reads, data.flank1Seed, bestn=1, nproc=args.nproc, outname=alignFile)
@@ -584,7 +584,7 @@
     
     if fl2Flag in data.stats["support"][2]:
         logging.debug("build flank2 %d" % fl2Flag)
-        alignFile = NamedTemporaryFile(prefix="f2con_", suffix=".m5", delete=False, dir=args.tempDir)
+        alignFile = NamedTemporaryFile(mode='w', prefix="f2con_", suffix=".m5", delete=False, dir=args.tempDir)
         alignFile.close(); alignFile = alignFile.name
         ALLTEMPFILES.append(alignFile)
         #blasr(data.flank2Reads, data.flank2Seed, bestn=1, nproc=args.nproc, outname=alignFile)
@@ -634,13 +634,13 @@
     """
     global ALLTEMPFILES
     data = alldata.stats
-    reads = NamedTemporaryFile(prefix="sol_", suffix=".fasta", delete=False, dir=args.tempDir)
+    reads = NamedTemporaryFile(mode='w', prefix="sol_", suffix=".fasta", delete=False, dir=args.tempDir)
     ALLTEMPFILES.append(reads.name)
     e1Seq = data["extendSeq1"]; e2Seq = data["extendSeq2"]
     reads.write(">%s\n%s\n>%s\n%s\n" % ("seq1", e1Seq, "seq2", e2Seq))
     reads.close()
     
-    alignFn = NamedTemporaryFile(prefix="sol_",suffix=".m5", delete=False, dir=args.tempDir)
+    alignFn = NamedTemporaryFile(mode='w', prefix="sol_",suffix=".m5", delete=False, dir=args.tempDir)
     ALLTEMPFILES.append(alignFn.name)
     blasr(reads.name, reads.name, nproc=args.nproc, outname=alignFn.name)
     aligns = M5File(alignFn)
@@ -688,9 +688,9 @@
     sent query, I'm going to pop all of the united reads onto this
     """
     global ALLTEMPFILES
-    alignFile = NamedTemporaryFile(prefix="uni_", suffix=".m5", delete=False, dir=args.tempDir).name
+    alignFile = NamedTemporaryFile(mode='w', prefix="uni_", suffix=".m5", delete=False, dir=args.tempDir).name
     ALLTEMPFILES.append(alignFile)
-    readFile = NamedTemporaryFile(prefix="uni_", suffix=".fasta", delete=False, dir=args.tempDir)
+    readFile = NamedTemporaryFile(mode='w', prefix="uni_", suffix=".fasta", delete=False, dir=args.tempDir)
     ALLTEMPFILES.append(readFile.name)
     
     input = FastqFile(inputFastq)
@@ -820,7 +820,7 @@
     
     preunitereads(supportFn, args)
     
-    onFlank = NamedTemporaryFile(prefix="onFlank_", suffix=".m5", delete=False, dir=args.tempDir)
+    onFlank = NamedTemporaryFile(mode='w', prefix="onFlank_", suffix=".m5", delete=False, dir=args.tempDir)
     ALLTEMPFILES.append(onFlank.name)
     onFlank.close()
     tailblasr(supportFn, flankFn, nproc=args.nproc, \
--- pbsuite/jelly/Support.py
+++ pbsuite/jelly/Support.py
@@ -8,6 +8,7 @@
 from pbsuite.utils.FileHandlers import *
 
 from networkx import Graph
+from functools import cmp_to_key
 
 USAGE = """Support.py <alignmentFile> <gapInfo> <outFile>
 
@@ -421,7 +422,7 @@
         
         newReads = self.layout(anchor, list(reads))
         
-        newReads.sort(cmp=lambda x,y: x.qstart < y.qstart)
+        newReads.sort(key=cmp_to_key(lambda x,y: x.qstart < y.qstart))
         return newReads
     
     """
@@ -483,7 +484,7 @@
                 regionHolder.qend = newGuys[-1].qend
                 ret.extend(newGuys)
         
-        ret.sort(cmp=lambda x,y: x.qstart < y.qstart)
+        ret.sort(key=cmp_to_key(lambda x,y: x.qstart < y.qstart))
         return ret
 
     def isDiscordant(self, alignment, tailAllowed=150):
@@ -703,7 +704,7 @@
         readName = alignmentGroup[0].qname
         logging.debug("looking at: " + readName + " for scaffold extension/unification")
         
-        alignmentGroup.sort(cmp=lambda x,y: x.qstart - y.qstart)
+        alignmentGroup.sort(key=cmp_to_key(lambda x,y: x.qstart - y.qstart))
         anchor = self.alignCon.getBestScore(alignmentGroup)
         
         flags = []
--- pbsuite/honey/bampie.py
+++ pbsuite/honey/bampie.py
@@ -331,7 +331,7 @@
         exit(1)

     logging.info("Extracting tails")
-    tailfastq = tempfile.NamedTemporaryFile(suffix=".fastq", delete=False, dir=args.temp)
+    tailfastq = tempfile.NamedTemporaryFile(mode='w',suffix=".fastq", delete=False, dir=args.temp)
     tailfastq.close(); tailfastq = tailfastq.name
     r, t, m = extractTails(bam, outFq=tailfastq, minLength=args.minTail)

@@ -342,7 +342,7 @@
         logging.info("No tails -- Exiting")
         exit(0)

-    tailmap = tempfile.NamedTemporaryFile(suffix=".sam", delete=False, dir=args.temp)
+    tailmap = tempfile.NamedTemporaryFile(mode='w',suffix=".sam", delete=False, dir=args.temp)
     tailmap.close(); tailmap = tailmap.name

     callBlasr(tailfastq, args.reference, args.params, args.nproc, tailmap)
@@ -375,7 +375,7 @@
             else:
                 logging.debug("Mapping %s" % ifile)
                 #Need to put this in a tempFile
-                outName = tempfile.NamedTemporaryFile(suffix="map%d.sam" % (c), \
+                outName = tempfile.NamedTemporaryFile(mode='w',suffix="map%d.sam" % (c), \
                                                      delete=False, dir=args.temp)
                 outName.close(); outName=outName.name
                 callBlasr(ifile, args.reference, args.params, args.nproc, outName)
--- pbsuite/honey/HSpots.py
+++ pbsuite/honey/HSpots.py
@@ -941,15 +941,15 @@
             #use the rest for cleaning

             #building consensus sequence
-            foutreads = NamedTemporaryFile(suffix=".fastq")
+            foutreads = NamedTemporaryFile(mode='w',suffix=".fastq")
             for id, i in enumerate(supportReads):
                 foutreads.write("@%d\n%s\n+\n%s\n" % (id, i[0], i[1]))
             foutreads.flush()
-            foutref = NamedTemporaryFile(suffix=".fasta")
+            foutref = NamedTemporaryFile(mode='w',suffix=".fasta")
             foutref.write(">%s:%d-%d\n%s" % (spot.chrom, start, end, refread[1]))
             foutref.flush()

-            alignOut = NamedTemporaryFile(suffix=".m5")
+            alignOut = NamedTemporaryFile(mode='w',suffix=".m5")
             logging.debug("making the contig....")
             #run it through phrap

@@ -984,12 +984,12 @@
             logging.debug("%s %d bp seq" % (conName, len(con.split('\n')[1])))

             #try improving consensus
-            conOut = NamedTemporaryFile(suffix=".fasta")
+            conOut = NamedTemporaryFile(mode='w',suffix=".fasta")
             conOut.write(con)
             #conOut.close()
             conOut.flush()

-            refOut = NamedTemporaryFile(suffix=".fasta")
+            refOut = NamedTemporaryFile(mode='w',suffix=".fasta")
             #j = reference.fetch(chrom, max(0, start-buffer), end+buffer)
             #fout = open("fuckme.ref.fasta",'w')
             #fout.write(j)
@@ -999,7 +999,7 @@
             refOut.flush()

             #map consensus to refregion
-            varSam = NamedTemporaryFile(suffix=".sam")
+            varSam = NamedTemporaryFile(mode='w',suffix=".sam")
             blasr(conOut.name, refOut.name, format="--sam", outname=varSam.name,\
                 consensus=False) #-- would this help?
                 #or what if I fed it through leftalign?
diff -r -u bin/m4pie.py bin_new/m4pie.py
--- pbsuite/jelly/m4pie.py
+++ pbsuite/jelly/m4pie.py
@@ -182,7 +182,7 @@
         exit(1)

     logging.info("Extracting tails")
-    tailfastq = tempfile.NamedTemporaryFile(suffix=".fasta", delete=False, dir=args.temp)
+    tailfastq = tempfile.NamedTemporaryFile(mode='w',suffix=".fasta", delete=False, dir=args.temp)
     tailfastq.close(); tailfastq = tailfastq.name
     logging.debug("Tail read tmp file %s " % (tailfastq))
     r, t, m = extractTails(aligns, reads, outFq=tailfastq, minLength=args.minTail)
@@ -195,7 +195,7 @@
         exit(0)

     logging.info("Mapping Tails")
-    tailmap = tempfile.NamedTemporaryFile(suffix=".m4", delete=False, dir=args.temp)
+    tailmap = tempfile.NamedTemporaryFile(mode='w',suffix=".m4", delete=False, dir=args.temp)
     tailmap.close(); tailmap = tailmap.name
     logging.debug("Read map tmp file %s " % (tailmap))
     mapTails(tailfastq, args.ref, nproc=args.nproc, out=tailmap, useSa=args.noSa)
diff -r -u bin/TGraf.py bin_new/TGraf.py
--- pbsuite/honey/TGraf.py
+++ pbsuite/honey/TGraf.py
@@ -699,7 +699,7 @@
             fout.write(str(clu) + "\t" + chrom + "\t" + j.toPrettyStr()+"\n")
             if args.fastq:
                 fastq = io.StringIO()
-                tfn = NamedTemporaryFile(suffix=".bam", delete=False).name
+                tfn = NamedTemporaryFile(mode='w',suffix=".bam", delete=False).name
                 align = pysam.Samfile(tfn, 'wb', template=bam)
                 for r in j.breads:
                     read = r.read
